src/Features/LanguageServer/Microsoft.CommonLanguageServerProtocol.Framework/IRequestContextFactory.cs

```diff
@@ -28,6 +28,6 @@ public interface IRequestContextFactory<TRequestContext>
     /// <param name="requestParam">The request parameters.</param>
     /// <param name="cancellationToken"></param>
     /// <returns>The <typeparamref name="TRequestContext"/> for this request.</returns>
-    /// <remarks>This method is called on the queue thread to allow context to be retrieved serially, without the posibility of race conditions from Mutating requests.</remarks>
-    Task<TRequestContext> CreateRequestContextAsync<TRequestParam>(IQueueItem<TRequestContext> queueItem, TRequestParam requestParam, CancellationToken cancellationToken);
+    /// <remarks>This method is called on the queue thread to allow context to be retrieved serially, without the possibility of race conditions from Mutating requests.</remarks>
+    Task<TRequestContext> CreateRequestContextAsync<TRequestParam>(IMethodHandler handler, IQueueItem<TRequestContext> queueItem, TRequestParam requestParam, CancellationToken cancellationToken);
```

Comment:
this is definitely used in Razor :(  Might need a second version of this interface.

Reply:
I was able to get this to work without changing this interface :)